FuelLabs/fuel-indexer
# enhancement: add support for tracking function calls in `Call` receipt

_Note: This PR is pointed at [`rashad/503-add-vec-support`](https://github.com/FuelLabs/fuel-indexer/tree/rashad/503-add-vec-support)._
## Changelog
- Add `abi::Call` struct
-- This struct substitutes the `param1` field in the `Call` receipt for a `fn_name: String` field so that users can see what function is being called
- Add a match structure for mapping function selectors to function names
- Adjust `Receipt::Call` match arm to generate an `abi::Call` struct
- Add test to trigger and index pure Sway function

## Testing Plan
Added new test to E2E test suite. CI should pass.

### Manual Testing Steps
0. Ensure your DB is cleared and that your WASM modules are up-to-date.
1. Run each of the following commands in a different terminal session:
- `cargo run --bin fuel-node`
- `cargo run --bin web-api`
- `cargo run --bin fuel-indexer -- run --manifest packages/fuel-indexer-tests/components/indices/fuel-indexer-test/fuel_indexer_test.yaml`
- `curl -X POST http://127.0.0.1:8000/pure_function`
2. You should see the data for `callentity` being indexed into the database.
